plugins/win-capture/window-helpers.c

```diff
@@ -383,6 +383,64 @@ static int window_rating(HWND window, enum window_priority priority,
 	return val;
 }
 
+static int window_rating_by_list(HWND window, const DARRAY(struct game_capture_picking_info) * games_whitelist)
+{
+	struct dstr cur_class = {0};
+	struct dstr cur_title = {0};
+	struct dstr cur_exe = {0};
+	int final_val = 0x7FFFFFFF;
+
+	if (!get_window_exe(&cur_exe, window))
+		return 0x7FFFFFFF;
+	get_window_title(&cur_title, window);
+	get_window_class(&cur_class, window);
+
+	
+	int i = 0;
+	while( i < games_whitelist->num )
+	{
+		int val = 0x7FFFFFFF;
+		bool class_matches = dstr_cmpi(&cur_class, games_whitelist->array[i].class.array) == 0;
+		bool exe_matches = dstr_cmpi(&cur_exe, games_whitelist->array[i].executable.array) == 0;
+		int title_val = abs(dstr_cmpi(&cur_title, games_whitelist->array[i].title.array));
+
+		/* always match by name with UWP windows */
+		bool uwp_window = dstr_cmpi(&games_whitelist->array[i].class, "Windows.UI.Core.CoreWindow") == 0;
+
+		if (uwp_window) {
```

Comment:
Why is it necessary do perform the following checks?

Reply:
This code taken from function that used to find manually selected window. 
As I understand that part should improve results in case of game written on "Universal Windows Platform". 

There is no perfect way to save exact window. as some games and non game may have same exe file( for example Minecraft is javaw.exe as many other programs), same for window class name. And window title can change from version to version or even for different users. 
So for games written on "Universal Windows Platform" where window class the same there is more strict check.